pkg/apiserver/registry/core/cloudprofile/strategy.go

```diff
@@ -96,3 +100,15 @@ func dropExpiredVersions(cloudProfile *core.CloudProfile) {
 		cloudProfile.Spec.MachineImages[i].Versions = validMachineImageVersions
 	}
 }
+
+func translateLegacyAccessRestrictionLabels(cloudProfile *core.CloudProfile) {
```

Comment:
If we removed it, we would break the current versions of dashboard and gardenctl.
I think, after some time has passed and these clients have adapted to the change, we can indeed remove the label and add another "cleanup step" here. I'll add some TODO statements.